src/epgsql_sock.erl

```diff
@@ -407,13 +435,22 @@ command_next(#state{current_cmd = PrevCmd,
                         results = []}
     end.
 
-
 setopts(#state{mod = Mod, sock = Sock}, Opts) ->
     case Mod of
         gen_tcp -> inet:setopts(Sock, Opts);
         ssl     -> ssl:setopts(Sock, Opts)
     end.
 
+-spec get_socket_active(pg_sock()) -> epgsql:socket_active().
+get_socket_active(#state{handler = H}) when H =/= on_replication ->
+    true;
+get_socket_active(#state{connect_opts = #{socket_active := Active}}) ->
+    Active;
+get_socket_active(#state{connect_opts = Opts}) when is_list(Opts) ->
+    proplists:get_value(socket_active, Opts, true);
```

Comment:
I know that typespec is confusing, but it is safe to assume `Opts` is always map. It is force-converted to map here:
https://github.com/epgsql/epgsql/blob/39d9662a22018faf0ec0372abc56c55f1420cf63/src/epgsql.erl#L201

And yep, I guess we should eventually have `epgsql:connect_opts_map()`, `epgsql:connect_opts_list()` and `epgsql:connect_opts() :: connect_opts_map() | connect_opts_list()`